db-service/lib/infer/index.js

```diff
@@ -380,6 +381,16 @@ function infer(originalQuery, model) {
     if (_.with)
       // consider UPDATE.with
       Object.values(_.with).forEach(val => inferQueryElement(val, false))
+    if (search) {
```

Comment:
Why do we need to infer the search elements? can't this be part of cqn4sql? I naively expected it there.

Reply:
Can't we do it roughly like this:

1. cqn4sql is called
2. as a very first step SELECT.search is rewritten to search function and appended to where clause
3. infer does its magic
4. cqn4sql builds join tree as usual

If we have an abstract function search that does everything search related, I'd like to keep the rest independent of search. hope it's not too weird and somehow understandable what I'd like to express :) if not, we should have a call :)

Reply:
I will try it.

There a couple of obstacles to overcome if we try to build a search term without inferring the query first.

1. there is no query target, can be solved by relying on Daniels infer by setting the prototype to `SELECT()` if a query has not been initialized with the CQL api.
2. columns are not yet resolved to query elements, hence there is no type information available at the column itself -> this becomes relevant for queries with "aggregations"

Reply:
okay, I made it work with `cds.infer`. However, I found a bug in Daniels implementation. We need `cap/cds/pull/4160` (hopefully with `cds@7.9.0`) before we can merge this.

Reply:
unfortunately, after upgrading to cds 7.9, [the HANA job is failing.](https://github.com/cap-js/cds-dbs/actions/runs/9063899197/job/24900854487)

@johannes-vogel @BobdenOs Do these errors look familiar to you? If not, I will have a closer look.